lib/package_provider/app/endpoints/packages.rb

```diff
@@ -25,7 +25,7 @@ class Packages < Base
 
         get '/download/:package_hash' do
           info = PackageProvider::CachedPackage.errors(params['package_hash'])
-          halt 404, info.to_json if info
+          halt 412, info.to_json if info
```

Comment:
definitely no 412, see: "The precondition given in one or more of the request-header fields evaluated to false..."
Maybee 422.